server/server_test.go

```diff
@@ -2121,6 +2121,239 @@ func TestStartPositionTimeDelta(t *testing.T) {
 	}
 }
 
+func TestStartPositionWithDurable(t *testing.T) {
+	s := RunServer(clusterName)
+	defer s.Shutdown()
+
+	sc := NewDefaultConnection(t)
+	defer sc.Close()
+	total := 100
+	for i := 0; i < total; i++ {
+		if err := sc.Publish("foo", []byte("msg")); err != nil {
+			t.Fatalf("Unexpected error on publish: %v", err)
+		}
+	}
+	ch := make(chan bool)
+	expected := uint64(1)
+	cb := func(m *stan.Msg) {
+		if m.Sequence == atomic.LoadUint64(&expected) {
+			sc.Close()
+			ch <- true
+		}
+	}
+	// Start a durable at first index
+	if _, err := sc.Subscribe("foo", cb,
+		stan.DeliverAllAvailable(),
+		stan.DurableName("dur"),
+		stan.MaxInflight(1)); err != nil {
+		t.Fatalf("Unexpected error on subscribe: %v", err)
+	}
+	// Wait for message to be received and connection to be closed.
+	if err := Wait(ch); err != nil {
+		t.Fatal("Did not get our message")
+	}
+	// Restart a connection
```

Comment:
Same. In some of the tests, I am closing the connection from the callback. The wait on channel gives me the assurance that both message and connection have been closed.